tests/js/client/server_parameters/legacy-sorting-cluster.js

```diff
@@ -0,0 +1,200 @@
+/* jshint globalstrict:false, strict:false, maxlen: 200 */
+/* global getOptions, assertEqual, assertTrue, assertFalse, arango */
+
+// //////////////////////////////////////////////////////////////////////////////
+// / DISCLAIMER
+// /
+// / Copyright 2014-2024 ArangoDB GmbH, Cologne, Germany
+// /
+// / Licensed under the Business Source License 1.1 (the "License");
+// / you may not use this file except in compliance with the License.
+// / You may obtain a copy of the License at
+// /
+// /     https://github.com/arangodb/arangodb/blob/devel/LICENSE
+// /
+// / Unless required by applicable law or agreed to in writing, software
+// / distributed under the License is distributed on an "AS IS" BASIS,
+// / WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// / See the License for the specific language governing permissions and
+// / limitations under the License.
+// /
+// / Copyright holder is ArangoDB GmbH, Cologne, Germany
+// /
+// / @author Max Neunhoeffer
+// //////////////////////////////////////////////////////////////////////////////
+
+// We need the following options for this test:
+//  --rocksdb.force-legacy-comparator true
+// This means that the old rocksdb comparator is used. This is necessary
+// since we want to test that the migration check detects bad data.
+
+if (getOptions === true) {
+  return {
+    'rocksdb.force-legacy-comparator': 'true'
+  };
+}
+
+let jsunity = require('jsunity');
+let internal = require('internal');
+let arangodb = require('@arangodb');
+let db = arangodb.db;
+
+// This introduces some values into a collection which will be sorted
+// incorrectly with the legacy sorting order. Note that we cannot do
+// this via JSON, since JavaScript has only double numbers. That is the
+// reason why we use the HTTP API directly and use awkward strings.
+
+function poisonCollection(databasename, collname, attrA, attrB) {
+  // 1152921504606846976 = 2^60 which does not completely fit into IEEE 754,
+  // therefore 2^60+1 and 2^60 are considered equal in the old comparator,
+  // if they are presented in different representations in vpack (int, uint,
+  // or double).
+  // Therefore, the following document with key "X" is considered smaller
+  // than the one with key "Y" in the legacy order, whereas in the
+  // correct lexicographic order document "X" is be larger.
+  let r = 
```

Comment:
`r` is assigned a value that is not used later. I suggest to remove the assignment here.

Reply:
I have checked the return code, which I forgot to add.